shell/platform/darwin/macos/framework/Source/FlutterEngineTest.mm

```diff
@@ -413,15 +400,13 @@ @interface FlutterEngine (Test)
   child1.tooltip = "";
   child1.child_count = 0;
   child1.custom_accessibility_actions_count = 0;
-  update_node_callback(&child1, (void*)CFBridgingRetain(engine));
-
-  FlutterSemanticsNode node_batch_end;
-  node_batch_end.id = kFlutterSemanticsNodeIdBatchEnd;
-  update_node_callback(&node_batch_end, (void*)CFBridgingRetain(engine));
 
-  FlutterSemanticsCustomAction action_batch_end;
-  action_batch_end.id = kFlutterSemanticsNodeIdBatchEnd;
-  update_action_callback(&action_batch_end, (void*)CFBridgingRetain(engine));
+  FlutterSemanticsUpdate update;
+  update.nodes_count = 2;
+  FlutterSemanticsNode nodes[] = {root, child1};
+  update.nodes = nodes;
+  update.custom_actions_count = 0;
+  update_semantics_callback(&update, (void*)CFBridgingRetain(engine));
```

Comment:
I believe this is a leak. The proc needs to be an ObjC proc for the capture to be collected after the callback is released.

Reply:
Thanks! It's not clear to me how `CFBridgingRetain` works, however, the documentation indicates this should be used when transferring ownership. That is indeed not the desired behavior, so I replaced the `CFBridingRetain` with `__bridge`.